src/braket/experimental/autoqasm/instructions/gates.py

```diff
@@ -0,0 +1,456 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+
+"""Quantum gates, unitary instructions, that apply to qubits.
+"""
+from braket.experimental.autoqasm import program
+
+from .qubits import _qubit
+
+
+def ccnot(control_0: int, control_1: int, target: int) -> None:
+    """CCNOT gate or Toffoli gate.
+
+    Args:
+        control_0 (QubitIdentifierType): Control qubit 0.
+        control_1 (QubitIdentifierType): Control qubit 1.
+        target (QubitIdentifierType): Target qubit.
+    """
+    qubits = [_qubit(control_0), _qubit(control_1), _qubit(target)]
+    oqpy_program = program.get_program_conversion_context().get_oqpy_program()
+    oqpy_program.gate(qubits, "ccnot")
```

Comment:
Is it worth reducing duplication a bit (and to isolate oqpy-related logic) by having a common implementation for all of these gates? For example, maybe this goes into `instructions.py`:
```
def _qubit_instruction(name: str, qubits: List[QubitIdentifierType], *args: Any):
    oqpy_program = program.get_program_conversion_context().get_oqpy_program()
    oqpy_program.gate([_qubit(q) for q in qubits], name, *args)
```
and then each gate (or other instruction such as `reset`) can be implemented as ~1 line:
```
def cphaseshift00(control: QubitIdentifierType, target: QubitIdentifierType, angle: float) -> None:
    """Controlled phase shift gate for phasing the \\|00> state.
    Args:
        control (QubitIdentifierType): Control qubit.
        target (QubitIdentifierType): Target qubit.
        angle (float): Rotation angle in radians.
    """
    _qubit_instruction("cphaseshift00", [control, target], angle)
```